test/models/repository_base.rb

```diff
@@ -76,5 +76,48 @@ def invalid_http_credentials(escape = false)
       end
       credentials
     end
+
+    def valid_docker_upstream_names
```

Comment:
These methods seem a little too specific to the tests to which they belong, but it's OK